bob/src/cluster/quorum.rs

```diff
@@ -83,28 +81,35 @@ impl Quorum {
 
     async fn delete_on_nodes(&self, key: BobKey) -> Result<(), Error> {
         debug!("DELETE[{}] ~~~DELETE LOCAL NODE FIRST~~~", key);
-        let res = delete_at_local_node(&self.backend, key).await;
-        let local_delete_ok = if let Err(e) = res {
-            error!("{}", e);
-            false
-        } else {
-            debug!("DELETE[{}] local node delete successful", key);
-            true
+        let (vdisk_id, disk_path) = self.mapper.get_operation(key);
+        let mut failed_nodes = vec![];
+        let mut total = 0;
+        if let Some(disk_path) = disk_path {
+            total += 1;
+            let res = delete_at_local_node(&self.backend, key, vdisk_id, disk_path, true).await;
+            if let Err(e) = res {
+                error!("{}", e);
+                failed_nodes.push(self.mapper.local_node_name().to_string());
+            }
         };
 
         debug!("DELETE[{}] ~~~DELETE TO REMOTE NODES~~~", key);
         let (errors, remote_count) = self.delete_at_remote_nodes(key).await;
-        let remote_ok_count = remote_count - errors.len();
-        if errors.len() > 0 || !local_delete_ok {
+        total += remote_count;
+        failed_nodes.extend(errors.iter().map(|o| o.node_name().to_string()));
+        if failed_nodes.len() > 0 {
```

Comment:
We should send a deletion request to aliens on ALL nodes without any conditions!